From 910847f45265aa1b0177dc4be4d165dc71c01b69 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 11 Oct 2023 16:01:13 +0200
Subject: [PATCH] glamor: Don't require EXT_gpu_shader4 unconditionally

It causes a shader compilation error on systems without EXT_gpu_shader4.

Fixes: ee107cd4
---
 glamor/glamor_render.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Index: xorg-server/glamor/glamor_render.c
===================================================================
--- xorg-server.orig/glamor/glamor_render.c
+++ xorg-server/glamor/glamor_render.c
@@ -237,7 +237,11 @@ glamor_create_composite_fs(glamor_screen
     const char *mask_fetch = "";
     const char *in;
     const char *header;
-    const char *header_norm = glamor_priv->glsl_version > 120 ? "#version 130\n" : "#version 120\n#extension GL_EXT_gpu_shader4 : require\n" GLAMOR_COMPAT_DEFINES_FS;
+    const char *header_norm = glamor_priv->glsl_version > 120 ?
+        "#version 130\n" :
+        glamor_priv->use_gpu_shader4 ?
+          "#version 120\n#extension GL_EXT_gpu_shader4 : require\n" GLAMOR_COMPAT_DEFINES_FS :
+          "#version 120\n" GLAMOR_COMPAT_DEFINES_FS;
     const char *header_es = glamor_priv->glsl_version > 100 ? "#version 300 es\n" : "#version 100\n" GLAMOR_COMPAT_DEFINES_FS;
     const char *dest_swizzle;
     GLuint prog;
